wptherml/exciton.py

```diff
@@ -0,0 +1,594 @@
+        )
+        self.c_vector = np.zeros(
+            (self.number_of_monomers, 1),  dtype=complex
+        )  # <== wavefunction coefficient vector
+        
+        # define cartesian coordinates of each monomer
+        self._compute_cartesian_coordinates()
+
+        # build exciton Hamiltonian
+        self.build_exciton_hamiltonian()
+
+        # Probably want to allow the user to specify an initial state!
+        # but right now just have the initial state with exciton localized on site 1
+        self.c_vector[0,0] = 1 + 0j
+        #self.density_matrix = np.dot(self.c_vector, np.conj(self.c_vector.T))
+        self.density_matrix = np.dot(self.c_vector, np.conj(self.c_vector.T))
+
+    def parse_input(self, args):
+        if "exciton_energy" in args:
+            self.exciton_energy = args["exciton_energy"]
+        else:
+            self.exciton_energy = 0.5
+
+        #############################################################
+        # going to change this to aggregate_shape 
+        #if "number_of_monomers" in args:
+        #    self.number_of_monomers = args["number_of_monomers"]
+        #else:
+        #    self.number_of_monomers = 2
+        #############################################################
+
+        if "aggregate_shape" in args:
+            self.aggregate_shape = args["aggregate_shape"]
+
+        # default shape is (2, 2, 1) -> 4 monomers total, 2 monomers displaced along x-axis and 2 along y-axis, all in plane (2D)
+        else:
+            self.aggregate_shape = (2, 2, 1)
+        
+        # user might accidentally make one element of the tuple zero
+        if self.aggregate_shape[0] == 0 or self.aggregate_shape[1] == 0 or self.aggregate_shape[2] == 0:
+            print(" Invalid shape!  All directions must have at least 1 layer!")
+            print(" Smallest possible shape (1,1,1) is a monomer - please check your input and try again")
+            exit
+
+        self.number_of_monomers = self.aggregate_shape[0] * self.aggregate_shape[1] * self.aggregate_shape[2]
+        # allow the user to specify the displacements along each axis (x, y, z) as a vector
+        if "displacement_vector" in args:
+            self.displacement_vector = args["displacement_vector"]
+
+        # default -> displacement along x = 35.47 a.u, displacement along y = 19.63 a.u., displacement along z = 19.63
+        else:
+            self.displacement_vector = [35.47, 19.63, 19.63]
+
+        if "transition_dipole_moment" in args:
+            self.transition_dipole_moment = args["transition_dipole_moment"]
+        else:
+            self.transition_dipole_moment = np.array([0, 0, 1])
+
+        if "refractive_index" in args:
+            self.refractive_index = args["refractive_index"]
+        else:
+            self.refractive_index = 1
+
+
+        self.wvlngth_variable = np.arange(0, 400.001, 0.01)
+
+    def _compute_cartesian_coordinates(self):
```

Comment:
Using the aggregate_shape and displacement_vector attributes, this method will fill out the cartesian coordinates for each monomer.